Source/WebGPU/WebGPU/XRSubImage.mm

```diff
@@ -66,6 +66,11 @@
 
 void XRSubImage::update(id<MTLTexture> colorTexture, id<MTLTexture> depthTexture, size_t currentTextureIndex, const std::pair<id<MTLSharedEvent>, uint64_t>& sharedEvent)
 {
+    RefPtr device = m_device.get();
+    auto* devicePtr = device.get();
+    if (!devicePtr)
+        return;
```

Comment:
Is there a need to store the raw pointer in a local variable? I would instead just do:

```
if (!device)
    return;
...
auto newTexture = Texture::create(..., *device);
...
```